### ISSUE TITLE
Two questions about the value loss and consistency loss.

### ISSUE THREAD

**Issue by niamorg** [nicklashansen/tdmpc2#80]
Hi, I have two very specific questions (maybe too precise and irrelevant for performance) about the losses of TD-MPC2 and TD-MPC:

1) In the papers TD-MPC, FOWL, TD-MPC2, it looks like the td-targets used for the Q-value loss are computed using:

$$\texttt{stop\\_grad}\left[ r_t + \gamma * Q\_\text{target}\left(z\_{t+1}, \pi(z\_{t+1})\right)\right]$$

where $`z_{t+1} = \text{latent\_dynamics}(z_t, a_t)`$ is the output of the latent rollout. But in the code, $`\text{encoder}(o_{t+1})`$ is used instead of $z_{t+1}$. Is there an explanation? Is it that $`\text{encoder}(o_{t+1})`$ is considered the "correct" latent for $o_{t+1}$, whereas $z_{t+1}$ is an approximation (improved via the consistency loss), and therefore  $`\text{encoder}(o_{t+1})`$ is chosen to have a better estimate of the td-target?

2) In TD-MPC and FOWL, the $z_\text{target}$ used in the consistency loss is $`\text{encoder}_\text{target}(o_{t+1})`$, whereas in TD-MPC2, it is $`\texttt{stop\_grad}\left[\text{encoder}(o_{t+1})\right]`$. Is this change to get rid of the target encoder network and/or because you noticed it had no impact on performance?

Also note that to compute the reward predictions, Q-value predictions and policy loss, it is the latent rollout $z_t$ which is used (altough not backpropagated in the policy loss). 

Thank you

**Comment by fernandopalafox** [comment]
Hey not sure if you figured this out, but this is why I think the code is like that: 
1. if you used the latents from the latent rollouts, compounding dynamics model error would leak into your TD targets. By using $`\text{encoder}(o_{t+1})`$ your only source of representation error is the encoder. 
2. Stopgrad prevents representation collapse without needing an extra encoder (and associated hyperparameters). [See this paper. ](https://arxiv.org/abs/2011.10566)

No idea about the impact on performance (i'm sure it's easy to test though), but hope this is useful.
